File: solution/402/solution.cpp
#include <stdio.h>
#include <vector>
#include <string>

class Solution {
public:
    string removeKdigits(string num, int k) {
        std::vector<int> S;
        std::string result = "";
        for(int i=0;i<num.length();i++){
            int number = num[i] - '0';
            while(k>0&&S.size()!=0&&number<S[S.size()-1]){
                S.pop_back();
                k--;
            }
            if(number!=0||S.size()>0){
                S.push_back(number);
            }
        }
        while(k>0&&S.size()>0){
            S.pop_back();
            k--;
        }
        for(int i=0;i<S.size();i++){
            result.append(1, '0' + S[i]);
        }
        if(result==""){
            result = "0";
        }
        return result;
    }
};

File: solution/402/solution.py
class Solution(object):
    def removeKdigits(self, num, k):
        """
        :type num: str
        :type k: int
        :rtype: str
        """
        S = []
        result = ""
        for x in num:
            x = int(x)
            while k > 0 and len(S) >0 and S[-1] > x:
                k -= 1
                del S[-1]
            if x != 0 or len(S) !=0:
                S.append(x)
        while k > 0 and len(S) > 0:
            k -= 1
            del S[-1]
        for x in S:
            result += str(x)
        if result == "":
            result = "0"
        return result